DuggaSys/courseed.js

```diff
@@ -37,31 +37,109 @@ function updateCourse()
-	localStorage.setItem('courseid', courseid);
-	localStorage.setItem('updateCourseName', true);
+	//check if token is set, if it is insert information into meta data
+	if(token){
+		$.ajax({
+			async: false,
+			url: "../DuggaSys/gitcommitService.php",
+			type: "POST",
+			data: {'githubURL':courseGitURL,'cid':cid,'token':token, 'action':'directInsert'},
+			success: function() { 
+				//Returns true if the data and JSON is correct
+				dataCheck = true;
+			},
+			error: function(data){
+				//Check FetchGithubRepo for the meaning of the error code.
+				switch(data.status){
+					case 403:
+						alert(data.status + " Error \nplease insert valid git key");
+						break;
+					case 422:
+						alert(data.responseJSON.message + "\nDid not create/update token");
+						break;
+					case 503:
+						alert(data.responseJSON.message + "\nDid not create/update token");
+						break;
+					default:
+						alert("Something went wrong with updating git token and git URL...");
+				}
+				dataCheck = false;
+			}
+		});
+	}
+	// Runs if token is not set.
+	else
+	{	
+		$.ajax({
+			async: false,
+			url: "../DuggaSys/gitcommitService.php",
+			type: "POST",
+			data: {'githubURL':courseGitURL,'cid':cid, 'action':'directInsert'},
+			success: function() { 
+				//Returns true if the data and JSON is correct
+				dataCheck = true;
+			},
+			error: function(data){
+				//Check FetchGithubRepo for the meaning of the error code.
+				switch(data.status){
+					case 403:
+						alert(data.status + " Error \nplease insert valid git key");
+						break;
+					case 422:
+						alert(data.responseJSON.message + "\nDid not create/update URL");
+						break;
+					case 503:
+						alert(data.responseJSON.message + "\nDid not create/update URL");
+						break;
+					default:
+						alert("Something went wrong with updating git URL..."+data.status);
+				}
+				dataCheck = false;
+			}
+		});
+	}
 
-	//Check if courseGitURL has a value
-	if(courseGitURL) {
-		//Check if fetchGitHubRepo returns true
-		if(fetchGitHubRepo(courseGitURL)) {
-			localStorage.setItem('courseGitHubRepo', courseGitURL);
-			//If courseGitURL has a value, display a message stating the update (with github-link) worked
-			alert("Course " + coursename + " updated with new GitHub-link!"); 
-			updateGithubRepo(courseGitURL, cid);
-		}
-		//Else: get error message from the fetchGitHubRepo function.
+	if(dataCheck)
+	{
+		// Show dialog
+		$("#editCourse").css("display", "none");
+		
+		// Updates the course (except the course GitHub repo. 
+		// Course GitHub repo is updated in the next block of code)
+		$("#overlay").css("display", "none");
+		AJAXService("UPDATE", {	cid : cid, coursename : coursename, visib : visib, coursecode : coursecode, courseGitURL : courseGitURL }, "COURSE");
+		localStorage.setItem('courseid', courseid);
+		localStorage.setItem('updateCourseName', true);
+
```

Comment:
Does all of this need to be saved to localstorage? This is not really part of the change request, just respond to this comment answering why.

Reply:
Gonna be honest. That part ain't ours. That's from the weakly branch so we didn't want to change anything if not needed